src/drivers/uavcan/sensors/gnss.cpp

```diff
@@ -121,6 +121,16 @@ UavcanGnssBridge::init()
 		_moving_baseline_data_pub_perf = perf_alloc(PC_INTERVAL, "uavcan: gnss: moving baseline data rtcm stream pub");
 	}
 
+	// If GPS1_ID is set, pre-allocate a channel by sending a blank report. This ensures that it becomes the first instance.
+	int32_t uavcan_gps1_id = 0;
+	param_get(param_find("UAVCAN_GPS1_ID"), &uavcan_gps1_id);
+
+	if (uavcan_gps1_id > 0) {
+		PX4_INFO("Allocating UAVCAN node %ld as GPS1 because of UAVCAN_GPS1_ID param", uavcan_gps1_id);
+		sensor_gps_s report{};
+		publish(uavcan_gps1_id, &report);
```

Comment:
There would be no GPS fix displayed in the top bar of QGC, so easy to see that something is wrong.

Reply:
But in this case, the ekf would switch to using the second gps instance.

Reply:
I was actually wrong above. The vehicle_gps_position module will actually not output a position unless the primary instance has at some point been available. This is likely to prevent arming without the primary GPS. So if this parameter is set to a non-existent ID no GPS would be selected. **This is, however, actually the behaviour we want!**

If the primary GPS has been available and then is disconnected, it would switch to the next gps instance, so if a sensor is lost during flight, we have redundancy.

Reply:
Here is the existing code that prevents fallback if no message has been received from primary GPS. Once again, I would argue that this is a good thing. 

https://github.com/aviant-tech/PX4-Autopilot/blob/9cfa0d25b215b2867f194124dda2be2ee1a91be8/src/modules/sensors/vehicle_gps_position/gps_blending.cpp#L131C1-L134C30